src/parameterizations/lateral/MOM_internal_tides.F90

```diff
@@ -502,6 +518,19 @@ subroutine propagate_int_tide(h, tv, cn, TKE_itidal_input, vel_btTide, Nb, dt, &
     enddo ; enddo
   enddo ; enddo ; enddo
 
+  ! loss from residual of reflection/transmission coefficients
+  if (CS%apply_residual_drag) then
+
+    do m=1,CS%nMode ; do fr=1,CS%nFreq ; do a=1,CS%nAngle ; do j=jsd,jed ; do i=isd,ied
+      ! implicit
+      CS%En(i,j,a,fr,m) = CS%En(i,j,a,fr,m) / (1.0 + dt * CS%TKE_residual_loss(i,j,a,fr,m) / &
```

Comment:
Please reformulate this expression to have only a single division, both for efficiency and clarity.  I can never remember whether F90 is left-associative or right associative, so that `A / B / C = A / (B*C)` or `A / B / C = A*C/B`, and I shouldn't have to.  Also, in many cases, I have found that refactoring such an expression actually makes it more robust and less prone to division by zero problem, and by explicitly refactoring to avoid extra divisions, we make it much more likely that the code compiled with debugging options or more aggressive optimization will give the same answers at roundoff.